### PR TITLE
custom-functions-runtime: Invocation / CancelableInvocation / StreamingInvocation

### PR DESCRIPTION
**Pull request by akrantz**
Adds CustomFunctions.Invocation with address property.
Renames CancelableHandler to CancelableInvocation, StreamingHandler to StreamingInvocation.
Updated doc comments.

---
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

### CODE REVIEW

**Comment by akrantz** [comment]
@Zlatkovsky @shaofengzhu Please review.

**Comment by akrantz** [comment]
@mscharlock Please review.

**Comment by mscharlock** [comment]
@akrantz Ok, this looks fine to me. What was the rationale behind the change - just consolidating all the invocations into the same category?

**Comment by akrantz** [comment]
@mscharlock We needed to also have the invocation parameter for non-cancelable/streaming functions so that they could receive the caller address if desired. If we followed the existing naming convention, the name "Handler" wouldn't be good. So the naming was updated to Invocation, CancelableInvocation, StreamingInvocation. I also improved the doc comments.

File: types/custom-functions-runtime/index.d.ts [hunk 1]
```diff
@@ -26,30 +29,67 @@ declare namespace CustomFunctions {
...
+         * Set the result for the custom function. May be called more than once.
+         */
+        setResult: (value: ResultType | Error) => void;
+    }
+
+    /**
+     * @beta
+     * @deprecated Use `StreamingInvocation<ResultType>` instead.
+     */
+    interface StreamingHandler<ResultType> extends StreamingInvocation<ResultType> {
```
Comment (Zlatkovsky) [inline types/custom-functions-runtime/index.d.ts]: Do you have a work item (filed internally or elsehow) to eventually remove these -- say, in a few months, after Script Lab and etc. have all been updated?

**Review by Zlatkovsky — APPROVED** [review]

**Comment by akrantz** [comment]
Please wait to merge this PR.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]
Marking as changes requested until @akrantz says it's ok to merge for bookkeeping purposes.

File: types/custom-functions-runtime/index.d.ts [hunk 2]
```diff
@@ -10,8 +13,8 @@ Copyright (c) Microsoft Corporation
 */
 
 /**
- * CustomFunctions namespace, used by Excel Custom Functions
  * @beta
+ * CustomFunctions namespace, used by Excel Custom Functions
```
[the remark this answers is not in the document]
Comment (mscharlock) [reply types/custom-functions-runtime/index.d.ts]: Apparently you can put that note wherever in the comments and the publishing system figures out where to put it

**Comment by mscharlock** [comment]
@akrantz I think we can go ahead and merge, I don't see any problems here. If you want to jump on that today that'd be awesome.

**Review by uniqueiniquity — APPROVED** [review]

**Comment by uniqueiniquity** [comment]
Congratulations on your first DefinitelyTyped contribution!
Thank you for being a part of the community!

**Pull request merged**